selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java

```diff
@@ -239,4 +307,77 @@ private void createFile(Path pathToRootContentDirectory, Path pathToFile) throws
   public String getFileContent(String pathToFile) throws IOException {
     return IOUtils.toString(ghRepo.getFileContent(pathToFile).read(), "UTF-8");
   }
+
+  public String getRepoSha() throws IOException {
+    return getReferenceToDefaultBranch().getObject().getSha();
+  }
+
+  public void addSubmodule(Path pathToRootContentDirectory, String submoduleName)
+      throws IOException, URISyntaxException, InterruptedException {
+
+    TestGitHubRepository submodule = new TestGitHubRepository(gitHubUsername, gitHubPassword);
+    submodule.addContent(pathToRootContentDirectory);
+    createSubmodule(submodule, submoduleName);
+    submodules.add(submodule);
+  }
+
+  private void createSubmodule(
+      TestGitHubRepository pathToRootContentDirectory, String pathForSubmodule)
+      throws IOException, URISyntaxException {
+    String submoduleSha = createTreeWithSubmodule(pathToRootContentDirectory, pathForSubmodule);
+
+    GHCommit treeCommit =
+        ghRepo.createCommit().tree(submoduleSha).message("Create submodule").create();
+
+    getReferenceToDefaultBranch().updateTo(treeCommit.getSHA1(), true);
+    createGitModulesFile(pathToRootContentDirectory, pathForSubmodule);
+  }
+
+  private boolean isGitmodulesFileExist() throws IOException {
+    return 0
+        < ghRepo
+            .getDirectoryContent("")
+            .stream()
+            .filter(item -> item.getName().equals(".gitmodules"))
+            .count();
+  }
+
+  private String createTreeWithSubmodule(
+      TestGitHubRepository targetRepository, String pathForSubmodule) throws IOException {
+    return ghRepo
+        .createTree()
+        .baseTree(this.getRepoSha())
+        .entry(
+            pathForSubmodule,
+            TreeElementMode.SUBMODULE.get(),
+            GitNodeType.COMMIT.get(),
+            targetRepository.getRepoSha(),
+            null)
+        .create()
+        .getSha();
+  }
+
+  private String getSubmoduleConfig(TestGitHubRepository targetRepository, String pathToSubmodule) {
+    String repoName = Paths.get(pathToSubmodule).getFileName().toString();
+    String repoUrl = targetRepository.getHtmlUrl() + ".git";
+    String modulePattern = "[submodule \"%s\"]\n\tpath = %s\n\turl = %s";
+
+    return String.format(modulePattern, repoName, pathToSubmodule, repoUrl);
+  }
+
+  private void createGitModulesFile(TestGitHubRepository targetRepository, String pathForSubmodule)
```

Comment:
Please, add link to the doc with explanation what .gitmodules file about. I would also recommend to use more general name of method like "setupSubmoduleConfig()".